components/CardForm/index.js

```diff
@@ -141,25 +182,64 @@ export default function CardForm({
             required
           />
           <StyledLabel htmlFor="date">Date: </StyledLabel>
-          <StyledInputField
-            type="date"
-            id="date"
-            name="date"
-            autoComplete="off"
-            min={getCurrentDate()}
-            defaultValue={existingActivityData?.date || ""}
-            required
-          />
+          <StyledDateWrapper>
+            <StyledInputField
+              type="text"
+              id="date"
+              name="date"
+              autoComplete="off"
+              value={
+                existingActivityData?.date
+                  ? existingActivityData.date
+                  : selectedDate
+                  ? format(selectedDate, "yyyy-MM-dd")
+                  : getCurrentDate()
+              }
+              readOnly
+              onClick={toggleDayPicker}
+              onChange={handleInputChange}
+              required
+            />{" "}
+            <StyledFiCalendar onClick={toggleDayPicker} />
+            {showDayPicker && (
+              <StyledDayPicker
+                onDayClick={handleDayClick}
+                selected={selectedDate}
+                modifiers={{ disabled: { before: new Date() } }}
+                locale={enGB}
+                format="yyyy-MM-dd"
+                formatDate={(date) =>
+                  format(date, "yyyy-MM-dd", { locale: enGB })
+                }
+              />
+            )}
+          </StyledDateWrapper>
           <StyledLabel htmlFor="time">Time: </StyledLabel>
-          <StyledInputField
-            type="time"
-            id="time"
-            name="time"
-            autoComplete="off"
-            min={getCurrentTime()}
-            defaultValue={existingActivityData?.time || ""}
-            required
-          />
+          <StyledDateWrapper>
+            <StyledInputField
+              type="text"
+              id="time"
+              name="time"
+              autoComplete="off"
+              defaultValue={existingActivityData?.time || ""}
```

Comment:
If you implemented the suggestion above then you could do the following. 

```suggestion
              defaultValue={time || ""}
```